skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFileService.kt

```diff
@@ -47,12 +40,11 @@ class SkaffoldFileService {
     fun isSkaffoldFile(file: VirtualFile): Boolean {
         with(file) {
             if (!isDirectory && fileType is YAMLFileType && isValid) {
-                val inputStream: InputStream = ByteArrayInputStream(contentsToByteArray())
-                inputStream.use {
-                    val scanner = Scanner(it)
-                    // consider this YAML file as Skaffold when first line contains proper API version
-                    return scanner.hasNextLine() &&
-                        SKAFFOLD_API_HEADER_PATTERN.matcher(scanner.nextLine()).find()
+                try {
+                    val skaffoldYaml = SkaffoldYamlConfiguration(this)
+                    return skaffoldYaml.apiVersion?.startsWith(SKAFFOLD_API_HEADER) == true
```

Comment:
no need for the `== true`

Reply:
ahh right. I guess the other way do deal with a `Boolean?` here would be to do `...startsWith(..) ?: false`. But what you have is clearer.